### ISSUE TITLE
[BUG] executions to context.exposeBinding in the frontend queue up and only reach the callback if the script is in the pause() state.

### ISSUE THREAD

**Issue by aruuuuu** [microsoft/playwright#12366]
**Context:**
- Playwright Version: 1.19.0
- Operating System: Mac
- Node.js version:v12.13.0
- Browser:Chromium
- Extra: [any specific details about your environment]

**Code Snippet**

Help us help you! Put down a short code snippet that illustrates your bug and
that we can run and debug locally. For example:

```java
import com.microsoft.playwright.*;
import org.junit.Test;

import java.nio.file.Paths;
import java.util.HashMap;
import java.util.Map;


public class TestPlayWright {
    @Test
    public void testPlaywrightExposeBindingCallbackFunctionality() throws Exception {
        try (Playwright playwright = Playwright.create()) {
            Browser browser = playwright.chromium().launch(new BrowserType.LaunchOptions().setHeadless(false).setSlowMo(50));
            Page page = browser.newPage();
            BrowserContext context = page.context();
            context.exposeBinding("clicked", (source, args) -> {
                        System.out.println("clicked!");
                        return null;
                    },
                    new BrowserContext.ExposeBindingOptions().setHandle(true));

            final String script =
                    "window.addEventListener('click', function (e) {  console.log('clicked'); console.log(e); window.clicked(e).bind(this); });";
            page.addInitScript(script);

            page.navigate("http://whatsmyuseragent.org/");
            page.screenshot(new Page.ScreenshotOptions().setPath(Paths.get("example.png")));

            for(int i = 0 ; i < 1_000_000;i++) {
                Thread.sleep(100);
            }
   
        }
    }
}


```

When I run the code above and try to run window.click() in the console, the call does not result in the callback of  context.exposeBinding("clicked", being called. Only once I go into the breakpoint in the for loop and run page.pause() do all those events flood the callback. This might be a bug since I would have expected the callback to always execute whenever window.click() is called in this case.

**Comment by mxschmitt** [comment]
This is expected, the general rule is to never use Thread.sleep and always use page.waitForTimeout instead, since when the Thread sleeps, no new events can come in anymore.

**Comment by mxschmitt** [comment]
Closing as part of the triage process since it seemed stale and answered. Please create a new issue with a detailed reproducible or feature request if you still face issues.
